backend/src/main/java/com/google/capstone/TextToSpeech.java

```diff
@@ -0,0 +1,64 @@
+package com.google.capstone;
+
+
+import java.io.IOException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.cloud.texttospeech.v1.AudioConfig;
+import com.google.cloud.texttospeech.v1.AudioEncoding;
+import com.google.cloud.texttospeech.v1.SsmlVoiceGender;
+import com.google.cloud.texttospeech.v1.SynthesisInput;
+import com.google.cloud.texttospeech.v1.SynthesizeSpeechResponse;
+import com.google.cloud.texttospeech.v1.TextToSpeechClient;
+import com.google.cloud.texttospeech.v1.VoiceSelectionParams;
+import com.google.protobuf.ByteString;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+
+
+
+@WebServlet(name = "TextToSpeech", value = "/api/v1/tts")
+public class TextToSpeech extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String textString = request.getParameter("text");
+    if (!textString.isEmpty()) {
+      try (TextToSpeechClient textToSpeechClient = TextToSpeechClient.create()) {
+        ByteString audioContents = generateAudio(textToSpeechClient, textString);
+        createFile(audioContents, response);
+      } catch (IOException e) {
+        response.sendError(500, "Unable to generate audio file.");
+      }
+    } else {
+      response.sendError(400, "Invalid text.");
+    }
+  }
+
+  private ByteString generateAudio(TextToSpeechClient textToSpeechClient, String textString) {
+    SynthesisInput input = SynthesisInput.newBuilder().setText(textString).build();
+    VoiceSelectionParams voice =
+            VoiceSelectionParams.newBuilder()
+                    .setLanguageCode("en-US")
+                    .setSsmlGender(SsmlVoiceGender.NEUTRAL)
+                    .build();
+    AudioConfig audioConfig =
+            AudioConfig.newBuilder().setAudioEncoding(AudioEncoding.MP3).build();
+    SynthesizeSpeechResponse ttsResponse =
+            textToSpeechClient.synthesizeSpeech(input, voice, audioConfig);
+    return ttsResponse.getAudioContent();
+  }
+
+  private void createFile(ByteString audioContents, HttpServletResponse response) throws IOException {
+    try (OutputStream out = new FileOutputStream("output.mp3")) {
```

Comment:
Using the same filename for all requests means that multiple concurrent requests could clobber each other. There are several ways to generate a unique filename (time-based, random, thread-safe atomic counter), and I'd recommend using one of them.

The easiest way in Java is probably to just use a random UUID: `UUID.randomUUID().toString()`
https://docs.oracle.com/javase/7/docs/api/java/util/UUID.html

If doing that creates issues and you can ensure your app will work properly, feel free to ignore me.

Reply:
This was more temporary to check that a file was being generated. In the following PR where I upload to cloud storage, the filename is based on an artwork's unique ID.